### PR TITLE
Update containers/image and containers/storage

### PR DESCRIPTION
**Pull request by nalind**
Update to currently-proposed changes for containers/image.  This is mainly a placeholder so that we can be extra-sure that those changes won't negatively impact cri-o.

### CODE REVIEW

**Comment by runcom** [comment]
This is fine but we need to wait for containers/image#305 still

**Comment by rhatdan** [comment]
LGTM other then the containers/image stuff.

**Comment by nalind** [comment]
Had to redo some of the bits of #654 that were reverted.

**Comment by runcom** [comment]
CI looks down, I'll re-trigger once it's back up

File: contrib/test/integration/build/kubernetes.yml [hunk 1]
```diff
@@ -43,9 +43,9 @@
       export CONTAINER_RUNTIME_ENDPOINT='{{ crio_socket }} --runtime-request-timeout=5m'
       export ALLOW_SECURITY_CONTEXT=","
       export ALLOW_PRIVILEGED=1
-      export DNS_SERVER_IP={{ ansible_eth0.ipv4.address }}
-      export API_HOST={{ ansible_eth0.ipv4.address }}
-      export API_HOST_IP={{ ansible_eth0.ipv4.address }}
+      export DNS_SERVER_IP={{ ansible_default_ipv4.address }}
+      export API_HOST={{ ansible_default_ipv4.address }}
+      export API_HOST_IP={{ ansible_default_ipv4.address }}
```
Comment (runcom) [inline contrib/test/integration/build/kubernetes.yml]: any reason to modify these?

File: contrib/test/integration/system.yml [hunk 2]
```diff
@@ -57,6 +58,22 @@
...
+  when: ansible_distribution in ['Fedora']
+
+- name: Add python-boto for RHEL and CentOS
+  package:
+    name: "{{ item }}"
+    state: present
+  with_items:
+   - python-boto
+  when: ansible_distribution in ['RedHat', 'CentOS']
+
```
Comment (runcom) [inline contrib/test/integration/system.yml]: why all these modifications here? where are those needed? right now these will only be picked with /test ami

File: contrib/test/integration/build/kubernetes.yml (hunk 1 above)
Comment (nalind) [reply contrib/test/integration/build/kubernetes.yml]: I happened to be trying to run the playbook against a system where the default network interface wasn't named "eth0".  This fixed it for me.

File: contrib/test/integration/system.yml (hunk 2 above)
Comment (nalind) [reply contrib/test/integration/system.yml]: Wait, we don't use this to set up the AMIs?  The package name is different between Fedora and RHEL/CentOS, and this made the task succeed for me.

File: contrib/test/integration/build/kubernetes.yml (hunk 1 above)
Comment (runcom) [reply contrib/test/integration/build/kubernetes.yml]: makes sense, I think there's no harm in having this then

File: contrib/test/integration/system.yml (hunk 2 above)
Comment (runcom) [reply contrib/test/integration/system.yml]: mmm where? I mean, current master is super ok test-wise, where was it failing for you?

Comment (runcom) [reply contrib/test/integration/system.yml]: > Wait, we don't use this to set up the AMIs?

we do but the AMI is cached, so, the job here is not running with this change, yet it pass right?

Comment (nalind) [reply contrib/test/integration/system.yml]: Ah, I was running the whole playbook on a fresh CentOS install.

Comment (runcom) [reply contrib/test/integration/system.yml]: well, I guess there's no harm in doing this as long as /test ami works, let's see

File: pkg/storage/image.go [hunk 3]
```diff
@@ -96,17 +171,27 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
...
-			img, err := ref.NewImage(systemContext)
+			img, err := ref.NewImageSource(systemContext)
 			if err != nil {
 				return nil, err
 			}
 			size := imageSize(img)
-			img.Close()
+			configDigest, err := imageConfigDigest(img, nil)
+			if err != nil {
+				return nil, err
```
Comment (nalind) [inline pkg/storage/image.go]: Doh, this doesn't Close() img.  Fixing.

**Comment by runcom** [comment]
failure in ami_rhel seems to be because of a runc bug, I'll create another PR to bump runc

**Pull request merged**